test/jest-light-runner/src/index.js

```diff
@@ -0,0 +1,56 @@
+import { Piscina } from "piscina";
+import supportsColor from "supports-color";
+import { MessageChannel } from "worker_threads";
+import os from "os";
+
+/** @typedef {import("@jest/test-result").Test} Test */
+
+const piscina = new Piscina({
+  filename: new URL("./worker-runner.js", import.meta.url).href,
+  maxThreads: os.cpus().length / 2,
```

Comment:
Test on my local machine is faster when I use the default `maxThreads` (`cpus().length * 1.5`) (29s vs 26s), but https://github.com/babel/babel/blob/893650130a9e678873345120f67c9316a3c590da/packages/babel-register/test/cache.js#L96-L106 is randomly failing. Guess `process.nextTick` has racing conditions?

Reply:
I don't see that failure :thinking:  Also, all the tests in a single file are run serially, so it shouldn't have any race conditions.

Btw, for me it takes way longer :grimacing: `33s` with `/ 2` vs `50s` with `* 1.5`.

Reply:
Here is an
<details>
<summary>error output</summary>

```
    - Expected  -  1
    + Received  + 45

    - Object {}
    + Object {
    +   "{\"assumptions\":{},\"sourceRoot\":\"/path/to/babel/packages/babel-register/test/fixtures/babelrc/\",\"cwd\":\"/path/to/babel/packages/babel-register/test/fixtures/babelrc\",\"babelrc\":false,\"sourceMaps\":false,\"caller\":{\"name\":\"@babel/register\"},\"filename\":\"/path/to/babel/packages/babel-register/test/fixtures/babelrc/es2015.js\",\"targets\":{},\"cloneInputAst\":true,\"configFile\":false,\"browserslistConfigFile\":false,\"passPerPreset\":false,\"envName\":\"undefined\",\"root\":\"/path/to/babel/packages/babel-register/test/fixtures/babelrc\",\"rootMode\":\"root\",\"plugins\":[],\"presets\":[]}:7.16.0:undefined": Object {
    +     "ast": null,
    +     "code": "const a = 1;",
    +     "map": null,
    +     "metadata": Object {},
    +     "mtime": 1637340521216,
    +     "options": Object {
    +       "assumptions": Object {},
    +       "ast": false,
    +       "babelrc": false,
    +       "browserslistConfigFile": false,
    +       "caller": Object {
    +         "name": "@babel/register",
    +       },
    +       "cloneInputAst": true,
    +       "configFile": false,
    +       "cwd": "/path/to/babel/packages/babel-register/test/fixtures/babelrc",
    +       "envName": "undefined",
    +       "filename": "/path/to/babel/packages/babel-register/test/fixtures/babelrc/es2015.js",
    +       "generatorOpts": Object {
    +         "comments": true,
    +         "compact": "auto",
    +         "filename": "/path/to/babel/packages/babel-register/test/fixtures/babelrc/es2015.js",
    +         "sourceFileName": "es2015.js",
    +         "sourceMaps": false,
    +         "sourceRoot": "/path/to/babel/packages/babel-register/test/fixtures/babelrc/",
    +       },
    +       "parserOpts": Object {
    +         "plugins": Array [],
    +         "sourceFileName": "/path/to/babel/packages/babel-register/test/fixtures/babelrc/es2015.js",
    +         "sourceType": "module",
    +       },
    +       "passPerPreset": false,
    +       "plugins": Array [],
    +       "presets": Array [],
    +       "root": "/path/to/babel/packages/babel-register/test/fixtures/babelrc",
    +       "rootMode": "root",
    +       "sourceMaps": false,
    +       "sourceRoot": "/path/to/babel/packages/babel-register/test/fixtures/babelrc/",
    +       "targets": Object {},
    +     },
    +     "sourceType": "module",
    +   },
    + }

      at file:/path/to/babel/packages/babel-register/test/cache.js:121:25

```
</details>

The test cache seems to be created by another test `test/index.js` but then read by `test/cache.js`, however I can't reproduce it consistently.

Reply:
> 33s with / 2 vs 50s with * 1.5.

Hmm, interesting. I think it is related to the Intel hyper-threading, which doubles the available processor count... Anyway the current setting is good enough for me, don't bother change that.

Reply:
Could you check if c385768f6107b85e3cc8f9bae877608b6f9323b3 solves the race condition?